### PR TITLE
4265 update tests to use new icon events

### LINKED ISSUE #4265: [Task] Update tests to use new icon events
See #4254

There's like 1000000 tests using `*-icon-cb` props, these all need updating to use `@click:*` events instead. The `NODE_ENV` check in util/console can then be removed.

### PR DESCRIPTION
**Pull request by ebisbe**
## Description
Fixes #4265 

## Motivation and Context
Help with some first issues

## How Has This Been Tested?
Upgrading tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

## Checklist:

- [ x] The PR title is no longer than 64 characters.
- [ x] The PR is submitted to the correct branch (`master` for bug fixes, `dev` for new features and breaking changes).
- [x ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have created a PR in the [documentation](https://github.com/vuetifyjs/vuetifyjs.com) with the necessary changes.

### CODE REVIEW

**Comment by ebisbe** [comment]
I made this branch from Dev. Should I have made it from Master? I can close it and redo it.

**Comment by KaelWD** [comment]
`git rebase --onto master HEAD~2 && git push -f`

File: src/util/console.ts [hunk 1]
```diff
@@ -35,7 +35,7 @@ export function consoleError (message: string, vm?: any, parent?: any): void {
 }
 
 export function deprecate (original: string, replacement: string, vm?: any, parent?: any) {
-  process.env.NODE_ENV === 'development' && consoleWarn(`'${original}' is deprecated, use '${replacement}' instead`, vm, parent)
```
Comment (johnleider) [inline src/util/console.ts]: Why is this removed?
